Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,254 @@
+        conceptsMergeBean.setWordnetIds(wordnetSet);
+        return conceptsMergeBean;
+    }
+
+    @Override
+    public void mergeConcepts(ConceptsMergeBean conceptsMergeBean, String userName)
+            throws LuceneException, IndexerRunningException, IllegalAccessException, DictionaryDoesNotExistException,
+            DictionaryModifyException {
+
+        deleteMergedConcepts(conceptsMergeBean.getConceptIds(), userName,
+                conceptsMergeBean.getSelectedConceptId().trim(), conceptsMergeBean.getSelectedListName());
+
+        if (conceptsMergeBean.getSelectedConceptId().trim().equals("")) {
+            // Add
+            ConceptEntry entry = new ConceptEntry();
+            getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+            conceptManager.addConceptListEntry(entry, userName);
+        } else {
+            // Update
+            ConceptEntry entry = conceptManager.getConceptEntry(conceptsMergeBean.getSelectedConceptId());
+            getConceptEntryFromConceptMergeBean(entry, conceptsMergeBean);
+            conceptManager.storeModifiedConcept(entry, userName);
+        }
+
+    }
+
+    private void deleteMergedConcepts(List<String> conceptIds, String userName, String conceptIdToNotDelete,
+            String selectedConceptList) throws LuceneException, IndexerRunningException, IllegalAccessException,
+            DictionaryDoesNotExistException, DictionaryModifyException {
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(conceptIdToNotDelete)) {
+                if (ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                            .getConceptTypeByConceptId(conceptId)) {
+                    // If its a wordnet concept, then create a wrapper and set
+                    // the delete flag to true. This is done in two steps
+                    // because changeevent object needs to be updated for
+                    // deletion correctly.
+                    String conceptWrapperId = createConceptWrapperById(conceptId, userName, selectedConceptList);
+                    conceptManager.deleteConcept(conceptWrapperId, userName);
+                } else {
+                    // If its a CCP concept, just delete it.
+                    conceptManager.deleteConcept(conceptId, userName);
+                }
+            }
+        }
+    }
+
+    private String createConceptWrapperById(String conceptId, String userName, String conceptListName)
+            throws IllegalAccessException, DictionaryDoesNotExistException, DictionaryModifyException, LuceneException,
+            IndexerRunningException {
+        ConceptEntry entry = conceptManager.getConceptEntry(conceptId);
+        entry.setConceptList(conceptListName);
+        // Not adding other details of merged concepts. Since its not relevant
+        // to add these details here.
+        return conceptManager.addConceptListEntry(entry, userName);
```

Comment:
you should still create a proper concept wrapper. We'll probably have restore, search, etc. functions for deleted concepts later on and for that we would need complete concepts.